app/models/service_order.rb

```diff
@@ -24,7 +26,10 @@ def create_order_name
     update_attributes(:name => "Order # #{id}") if name.blank?
   end
 
-  def checkout
+  def checkout(checkout_immediately = false)
+    unless checkout_immediately
+      raise "Invalid operation [checkout] for Service Order in state [#{state}]" if state == STATE_ORDERED
```

Comment:
Wait a minute... you want to raise if they are trying to checkout an already ordered cart, right?  So it should really be:
```ruby
raise "Invalid operation [checkout] for Service Order in state [#{state}]" if checkout_immediately && state == STATE_ORDERED
```